src/compiler/binder.ts

```diff
@@ -3289,7 +3289,7 @@ function createBinder(): (file: SourceFile, options: CompilerOptions) => void {
             return;
         }
         const rootExpr = getLeftmostAccessExpression(node.left);
-        if (isIdentifier(rootExpr) && lookupSymbolForName(container, rootExpr.escapedText)?.flags! & SymbolFlags.Alias) {
+        if (isIdentifier(rootExpr) && lookupSymbolForName(container, rootExpr.escapedText)!.flags & SymbolFlags.Alias) {

```

Comment:
[@typescript-eslint/no-non-null-asserted-optional-chain](https://typescript-eslint.io/rules/no-non-null-asserted-optional-chain): _"Optional chain expressions can return undefined by design - using a non-null assertion is unsafe and wrong."_

The `?.` followed by a `!` indicates the value must exist, so it's more succinct to use a single `!`.

Reply:
that looks unsafe... previously it resulted in `undefined & SymbolFlags.Alias` which is just 0, now it would crash at runtime

Reply:
Ha, you're right - that explains the test failures. I'll move the rule to the _todo_ section in the ESLint config.